client/src/components/ui/dropdown.tsx

```diff
@@ -0,0 +1,180 @@
+  id,
+  children,
+  onClick,
+  style,
+  caretWidth,
+}: DropdownButtonProps) {
+  return (
+    <button
+      id={id}
+      type="button"
+      className={style}
+      aria-haspopup={true}
+      aria-expanded={true}
+      onClick={onClick}
+    >
+      <div className="flex w-full items-center justify-between">
+        {children}
+
+        <div
+          className="flex items-center justify-end"
+          style={{ width: `${caretWidth}px` }}
+        >
+          <DropdownIcon strokeColour="penni-text-regular-light-mode" />
+        </div>
+      </div>
+    </button>
+  );
+}
+
+export function DropdownMenu({ menuId, options, onChange }: DropdownMenuProps) {
+  return (
+    <div className="relative">
+      <div className="h-auto w-full">
+        <ul
+          className="absolute left-0 z-10 mt-px flex w-full origin-top-right flex-col rounded-penni-card bg-penni-background-input-light-mode px-2 py-2 text-center shadow-lg focus:outline-none"
+          role="menu"
+          aria-orientation="vertical"
+          aria-labelledby={menuId}
+          tabIndex={-1}
+        >
+          {options.map((option, index) => (
+            <li
+              key={index}
+              value={option}
+              className={`${valueStyle} select-none rounded-penni-border px-2 py-2 hover:cursor-pointer hover:bg-penni-grey-inactive`}
+              role="menuitem"
+              tabIndex={-1}
+              onClick={() =>
+                onChange({
+                  target: { value: option },
+                } as React.ChangeEvent<HTMLEventTargetElement>)
+              }
+            >
+              {option}
+            </li>
+          ))}
+        </ul>
+      </div>
+    </div>
+  );
+}
+
+/**
+   * Dropdown component in bidders side market page with a list of options.
+   
+   * @param props - The properties for the DropdownInput component.
+   * @param props.value - Value of the selected option, tracked by `react.setState()`.
+   * @param props.options - List of possible option for that category, displayed in dropdown.
+   * @param props.onChange - Called when the selected value changes, should update `value`.
+   * @param props.label - Label for the field, display "All ...".
+   *
+   * @returns dropdown component to select different categories.
+   *
+   * @example
+   * // Example usage:
+   * // initial state is '', display all tasks, them use onChange function to get selected category that can be used to send query to database
+   * const [selectedValue, setSelectedValue] = useState('');
+   * <MarketDropdown
+   *   value={selectedValue}
+   *   options={['C', 'Python', 'HTML']}
+   *   onChange={(e) => setSelectedValue(e.target.value)}
+   *   label="All Language"
+   * />
+   */
+export function MarketDropdown({
+  value,
+  options,
+  onChange,
+  label,
+}: MarketDropdownProps) {
+  const [menuId] = useState(uniqueId());
+  const [isExpanded, setExpanded] = useState(false); // isSelect in text inputs
```

Comment:
Belongs in `<DropdownButton>`, also remove the comment since this is not in `inputs.tsx` anymore